external/storm-kafka/src/jvm/org/apache/storm/kafka/ExponentialBackoffMsgRetryManager.java

```diff
@@ -86,15 +94,23 @@ public Long nextFailedMessageToRetry() {
     }
 
     @Override
-    public boolean shouldRetryMsg(Long offset) {
+    public boolean shouldReEmitMsg(Long offset) {
         MessageRetryRecord record = this.records.get(offset);
         return record != null &&
                 this.waiting.contains(record) &&
                 System.currentTimeMillis() >= record.retryTimeUTC;
```

Comment:
I would be nice to log a WARN message when we are dropping messages so users have a way of knowing when/why it is happening.